File: Classes/MainScene.cpp
//
//  MainScene.cpp
//  com.goofypapa.GoofypapaTest-mobile
//
//  Created by 吴思 on 2018/12/4.
//

#include "MainScene.h"
#include "JsScene.h"

USING_NS_CC;

bool MainScene::init( void )
{
    if( !Scene::init() )
    {
        return false;
    }
    
    auto t_visibleSizeHalf = Director::getInstance()->getVisibleSize() * 0.5f;
    Vec2 t_origin = Director::getInstance()->getVisibleOrigin();
    
    auto t_itemTest = MenuItemFont::create( "Test", []( Ref * p_target ){
        Director::getInstance()->replaceScene( JsScene::create( "Test/main.js" ) );
    } );
    
    t_itemTest->setPosition( Vec2( t_origin.x + t_visibleSizeHalf.width, t_origin.y + t_visibleSizeHalf.height * 1.5f ) );
    
    auto t_itemAstronomy = MenuItemFont::create( "Astronomy", []( Ref * p_target ){
        Director::getInstance()->replaceScene( JsScene::create( "Astronomy/main.js" ) );
    } );
    t_itemAstronomy->setPosition( Vec2( t_origin.x + t_visibleSizeHalf.width, t_origin.y + t_visibleSizeHalf.height ) );
    
    auto t_itemABC = MenuItemFont::create( "ABC", []( Ref * p_target ){
        Director::getInstance()->replaceScene( JsScene::create( "ABC/main.js" ) );
    } );
    t_itemABC->setPosition( Vec2( t_origin.x + t_visibleSizeHalf.width, t_origin.y + t_visibleSizeHalf.height * 0.5f ) );
    
    
    auto t_menu = Menu::create( t_itemTest, t_itemAstronomy, t_itemABC, NULL );
    t_menu->setPosition( Vec2::ZERO );
    addChild( t_menu );
    
    return true;
}

File: App/Test/mainScene.js

MainSceneLayer = cc.Layer.extend({
    ctor: function(){

        this._super();

        var visibleSize = cc.director.getVisibleSize();
        var visibleOrigin = cc.director.getVisibleOrigin();

        console.log( "------->" + JSON.stringify( visibleSize ) );
        console.log( "------->" + JSON.stringify( visibleOrigin ) );

        var sprite = new cc.Sprite( "HelloWorld.png" );
        var spriteSize = sprite.getContentSize();
        sprite.setPosition( visibleOrigin.x + visibleSize.width - spriteSize.width / 2, visibleOrigin.y + spriteSize.height / 2 );
        this.addChild( sprite );

        console.log( "==============" + JSON.stringify( spriteSize ) );

        var btn = new cc.MenuItemFont("Go Sub Scene", function(sender) {
            cc.director.replaceScene( new SubScene() );
        }, this);
        var menu = new cc.Menu( btn );
        menu.setPosition( visibleOrigin.x + visibleSize.width / 2, visibleOrigin.y + visibleSize.height / 2 );
        this.addChild( menu );
    }
});

var MainScene = {
    create: function(){
        var scene = new cc.Scene();
        var layer = new MainSceneLayer();
        scene.addChild( layer );
        return scene;
    }
};
